### PR TITLE
Propagate the surplus capturing JIT-order owners to the solvers

### PR DESCRIPTION
**Pull request by Mateo-mro**
# Description
The solvers would need to know the JIT-order surplus capturing addresses (see e.g. https://github.com/cowprotocol/services/issues/2715#issuecomment-2130855593). For that the owner addresses are propagated to the solvers in the auction json.

# Changes
Propagate the list of surplus capturing JIT-order owners to the solvers in the auction json.

## How to test
1. Regression tests

### CODE REVIEW

File: crates/driver/src/infra/solver/dto/auction.rs [hunk 1]
```diff
@@ -349,8 +355,8 @@ struct Order {
     valid_to: u32,
     kind: Kind,
     #[serde(skip_serializing_if = "Option::is_none")]
-    receiver: Option<H160>,
-    owner: H160,
+    receiver: Option<eth::H160>,
+    owner: eth::H160,
```
Comment (Mateo-mro) [inline crates/driver/src/infra/solver/dto/auction.rs:359]: Removed the `H160` import and make it all as `eth::H160` to keep consistency

File: crates/solvers/openapi.yml [hunk 2]
```diff
@@ -658,6 +659,12 @@ components:
             by the caller.
           allOf:
             - $ref: "#/components/schemas/DateTime"
+        surplusCapturingJitOrderOwners:
+          type: array
+          items:
+            $ref: "#/components/schemas/Address"
+          description: |
+            List of addresses on whose surplus will count towards the objective value of their solution (unlike other orders that were created by the solver).
```
Comment (squadgazzz) [inline crates/solvers/openapi.yml:667]: We need the same for the `gnosis/solvers` repo

**Review by squadgazzz — APPROVED** [review]

**Review by sunce86 — COMMENTED** [review]
Also solvers-dto crate needs to be updated

File: crates/solvers/openapi.yml (hunk 2 above)
Comment (Mateo-mro) [reply crates/solvers/openapi.yml:667]: @squadgazzz good point! https://github.com/gnosis/solvers/pull/28

**Review by fleupold — APPROVED** [review]
You probably know, but this will also require a change in https://github.com/gnosis/solvers/blob/ca55c3cca15630891f653bca077a8a0c6b364c39/Cargo.toml#L49

**Comment by Mateo-mro** [comment]
> You probably know, but this will also require a change in https://github.com/gnosis/solvers/blob/ca55c3cca15630891f653bca077a8a0c6b364c39/Cargo.toml#L49

@fleupold yup, the PR is ready, I just need to merge this first and then push the tag.

**Pull request merged**